strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPath.java

```diff
@@ -306,6 +310,14 @@ public RepositoryPath wrap(Path path)
     {
         return new RepositoryPath(path, fileSystem);
     }
+
+    public boolean isOld()
+            throws IOException
+    {
+        final Instant twentyFourHoursAgo = Instant.now().minus(24, ChronoUnit.HOURS);
```

Comment:
I think a better check would be this: 24 hrs should actually mean "the files expire at midnight". This is, for example, how Maven knows whether to check for a newer version of a timestamped snapshot (as the default update policy is `daily`, meaning: "if there is no cached resource, get it from the remote and if a consecutive request is received later on, don't get the artifact before midnight". I know that some proxy repositories follow this approach as well and I think we should do it too.

This should be configurable via the `strongbox.xml` file. Furthermore, we need to have an option to define the caching period for the metadata (and each layout provider will have to expose a list of path patterns that it considers metadata). I believe this is quite important.

Reply:
Also this attribute probably should be placed into `RepositoryFileAttributes` so the implementation will be in `AbstractLayoutProvider.getRepositoryFileAttributes`.

Reply:
Also, I believe the expiration interval should be configurable in minutes (for now; later on we might further improve this).